contracts/exchangeIssuance/ExchangeIssuanceLeveragedForCompound.sol

```diff
@@ -0,0 +1,1346 @@
+     * @param _premium             Premium / Interest that has to be returned to the lending pool on top of the loaned amount
+     * @param _decodedParams       Struct containing token addresses / amounts to perform redemption
+     */
+    function _performRedemption(
+        address _debtToken,
+        uint256 _debtTokenAmountNet,
+        uint256 _premium,
+        DecodedParams memory _decodedParams
+    )
+    internal
+    {
+        // Redeem set using debt tokens obtained from flashloan
+        _redeemSet(
+            _decodedParams.setToken,
+            _decodedParams.setAmount,
+            _decodedParams.originalSender
+        );
+
+        _withdrawFromCompound(
+            _decodedParams.leveragedTokenData.collateralToken,
+            _decodedParams.leveragedTokenData.collateralAmount
+        );
+        // Obtain debt tokens required to repay flashloan by swapping the underlying collateral tokens obtained in withdraw step
+        uint256 collateralTokenSpent = _swapCollateralForDebtToken(
+            _debtTokenAmountNet + _premium,
+            _debtToken,
+            _decodedParams.leveragedTokenData.collateralAmount,
+            _decodedParams.leveragedTokenData.collateralToken,
+            _decodedParams.collateralAndDebtSwapData
+        );
+        // Liquidate remaining collateral tokens for the payment token specified by user
+        uint256 amountOutputToken = _liquidateCollateralTokens(
+            collateralTokenSpent,
+            _decodedParams.setToken,
+            _decodedParams.setAmount,
+            _decodedParams.originalSender,
+            _decodedParams.paymentToken,
+            _decodedParams.limitAmount,
+            _decodedParams.leveragedTokenData.collateralToken,
+            _decodedParams.leveragedTokenData.collateralAmount  - 2*ROUNDING_ERROR_MARGIN,
+            _decodedParams.paymentTokenSwapData
+        );
+        require(amountOutputToken >= _decodedParams.limitAmount, "ExchangeIssuance: INSUFFICIENT OUTPUT AMOUNT");
+    }
+
+
+    /**
+    * Returns the collateral / debt token addresses and amounts for a leveraged index
+    *
+    * @param _setToken              Address of the SetToken to be issued / redeemed
+    * @param _setAmount             Amount of SetTokens to issue / redeem
+    * @param _isIssuance            Boolean indicating if the SetToken is to be issued or redeemed
+    *
+    * @return Struct containing the collateral / debt token addresses and amounts
+    */
+    function _getLeveragedTokenData(
+        ISetToken _setToken,
+        uint256 _setAmount,
+        bool _isIssuance
+    )
+    internal
+    view
+    returns (LeveragedTokenData memory)
+    {
+        address[] memory components;
+        uint256[] memory equityPositions;
+        uint256[] memory debtPositions;
+        address _collateralToken;
+        LeveragedTokenData memory _leveragedTokenData;
+        uint256 requiredCTokenAmount;
```

Comment:
Maybe rename this variable just `cTokenAmount` since it isn't really "required" in the redemption case, rather it is the expected amount received upon redemption.

Reply:
addressed in 2bc43dfd123d9c826c22e1931048af595bb3ebc3